blockchain/state_manager/src/lib.rs

```diff
@@ -234,4 +244,31 @@ where
             )),
         }
     }
+
+    pub fn lookup_id(&self, addr: &Address, ts: &Tipset) -> Result<Address, Error> {
+        let state_tree = StateTree::new_from_root(self.bs.as_ref(), ts.parent_state())?;
+        state_tree.lookup_id(addr).map_err(Error::State)
+    }
+
+    pub fn market_balance(&mut self, addr: &Address, ts: &Tipset) -> Result<MarketBalance, Error> {
+        let market_state: market::State =
+            self.load_actor_state(&*STORAGE_MARKET_ACTOR_ADDR, ts.parent_state())?;
+
+        let new_addr = self.lookup_id(addr, ts)?;
+
+        let out = MarketBalance {
+            escrow: {
+                let et = BalanceTable::from_root(self.bs.as_ref(), &market_state.escrow_table)
+                    .map_err(|_x| Error::State("Failed to build Escrow Table".to_owned()))?;
```

Comment:
```suggestion
                    .map_err(|_x| Error::State("Failed to build Escrow Table".to_string()))?;
```

Reply:
any reason why you suggest this? `to_string` just calls `to_owned` and I think it's pretty opinionated which is more readable

Reply:
I guess it is pretty opinionated!, that is true. For me I just think to_string makes it explicit that I am changing from a type of &Str to a type of String.

Reply:
I also very slightly prefer `to_string` but I think this is a lost cause without a linter that enforces it 😛 There are just too many ways to do this conversion

Reply:
yeah sure thing, the default throughout the existing code is `to_owned` just because to me it was more explicit (since the to_string method usually comes with other conversion logic from types that use the `FromStr` trait) but happy to start using `to_string` as a default moving forward as it doesn't matter